### PR TITLE
Clean up repo, use CPM for dependency management

### PR DESCRIPTION
**Pull request by maztheman**
- Update CMake version to 3.21
- Make this repo able to be pulled in as a dependency via CMake
- Use new compiler and environment detection (from range-v3)
- Use CPM for dependencies
- Fix Github actions to work on Clang 10,11,12,14

### CODE REVIEW

**Comment by maztheman** [comment]
vcpkg install in the github action is no longer really needed..but left it in for now

**Comment by maztheman** [comment]
I didn't mean to add that second commit, oops.

**Review by glynos — APPROVED** [review]

**Comment by glynos** [comment]
> I didn't mean to add that second commit, oops.

Can you rebase and remove it? Then I'll merge

**Comment by maztheman** [comment]
done.

File: .github/workflows/skyr-url-ci.yml [hunk 1]
```diff
@@ -451,8 +475,9 @@ jobs:
               -D skyr_BUILD_DOCS=OFF
               -D skyr_BUILD_EXAMPLES=OFF
               -D skyr_BUILD_V2=${BUILD_V2}
+              -D skyr_BUILD_WITH_LLVM_LIBCXX=${USE_LLVM}
+              -D SKY_VERBOSE_BUILD=ON
```
[the remark this answers is not in the document]
Comment (maztheman) [reply .github/workflows/skyr-url-ci.yml:479]: Looks like the new variables I put in has prefix SKY, but that doesn't seem to be consistent with everything else.

**Pull request merged**